core/src/main/scala/spire/math/ULong.scala

```diff
@@ -64,9 +64,13 @@ class ULong(val signed: Long) extends AnyVal {
     if (signed >= 0L) signed.toString
     else (ULong.LimitAsBigInt + signed).toString
 
+  // TODO: replace these with `===` and `=!=`? Don't we get `==` and `!=` for free with value classes?
```

Comment:
From this REPL test (`scala 2.11.7`), it **inconclusively** seems to me that the *free* `==` and `!=` accept an argument of `Any`, not `ULong`, as you wrote in lines 68-69.

I admit that I know nothing about value classes other than reading this article - http://docs.scala-lang.org/overviews/core/value-classes.html.

```
scala> class BoolWrapper(val underlying: Boolean) extends AnyVal
defined class BoolWrapper

scala> class Wrapper(val underlying: Int) extends AnyVal
defined class Wrapper

scala> new Wrapper(5) == new BoolWrapper(true)
<console>:15: warning: Wrapper and BoolWrapper are unrelated: they will never compare equal
       new Wrapper(5) == new BoolWrapper(true)
                      ^
res6: Boolean = false

scala> new Wrapper(5) != new BoolWrapper(true)
<console>:15: warning: Wrapper and BoolWrapper are unrelated: they will always compare unequal
       new Wrapper(5) != new BoolWrapper(true)
                      ^
res7: Boolean = true
```

Also, I read this som-snytt [answer](http://stackoverflow.com/a/18585096/409976) on `why-operator-and-equals-behave-differently-for-values-of-anyval-in-scala`. However, I don't entirely understand it.